bookwyrm/templates/shelf/shelf.html

```diff
@@ -189,6 +192,19 @@ <h2 class="title is-3">
                         {% include 'snippets/stars.html' with rating=book.rating %}
                     </td>
                     {% endif %}
+                    {% if shelves|length > 0 %}
+                    <td data-title="{% trans "Shelves" %}">
+                        {% spaceless %}
+                        {% for shelf in shelves %}
+                        {% if shelf in book.shelves.all %}
```

Comment:
I think `book.shelves` will produce all shelves for all users, rather than the shelf related to the current user

Reply:
Oh, we're actually both sort of right -- `book.shelves.all` _does_ produce all shelves for all users, but because it's checking if a `Shelf` object is present, and a Shelf is tied to a user, it still displays correctly, so this does work exactly how you intended. 

My new qualm is that on a large instance, that's going to be a costly lookup, since there are a lot of shelves. It may be that assigning `book.shelves.all` to a variable using `{% set ... %}`, so that the query is only called once rather than once per shelf, will be enough mitigation?